ParquetClassLibrary/Sandbox/Parquets/AllParquets.cs

```diff
@@ -27,35 +30,85 @@ public static class AllParquets
         /// </exception>
         public static T Get<T>(ParquetID in_ID) where T : ParquetParent
         {
-            return (T)_parquetDefinitions[in_ID];
+            return (T)ParquetDefinitions[in_ID];
         }
 
         /// <summary>
-        /// Loads all parquets definitions.
+        /// Adds the given parquet to the collection of cannonical parquets.
+        /// This allows parquets to be created at run-time.
         /// </summary>
-        public static void DeserializeAllParquets()
+        /// <param name="in_parquet">The parquet being defined.</param>
+        /// <returns><c>true</c> if the parquet was added successfully; <c>false</c> otherwise.</returns>
+        public static bool Put(ParquetParent in_parquet)
         {
-            /* TODO: Ensure we have no duplicate values.
-                if (KnownIDs.Contains(value))
-                {
-                    Error.Handle($"Tried to create duplicate parquet with ID {value}.");
-                }
-                else
-                {
-                    KnownIDs.Add(value);
-                    _id = value;
-                }
-             */
-            // TODO Implement me!
-            // IDEA Use CSV serializer.
+            var isNew = !ParquetDefinitions.ContainsKey(in_parquet.ID);
+
+            if (isNew)
+            {
+                ParquetDefinitions[in_parquet.ID] = in_parquet;
```

Comment:
comment: I'm now realizing I've always used ParquetDefinitions.Add(in_parquet.ID, in_parquet) rather than assigning to the dictionary. I think that comes from being a C# developer first, but other languages (like Javascript) work like this.

Reply:
As afar as I know both are okay.  For me, this more closely reflects my intent, so I try to use it when I think of it.